### ISSUE TITLE
Cannot re-add an previously existing user and change their name at the same time

### ISSUE THREAD

**Issue by raymond-ensemble** [adobe-apiplatform/user-sync.py#114]
**Steps**
1. Sync and add a new user from AD with a first name: "Name1"
2. Go to the dashboard and delete the user
3. In AD, change that user's name from "Name1" to "Name2"
4. Sync the user again with --update-user-info

**Results**
- After the second sync, it would display "Name1" as the first name on the dashboard. Only after you run --update-user-info again would the name be updated to "Name2".
- It seems like you cannot re-add a previously existing user and change their name at the same time.

**Comment by adobeDan** [comment]
So I need the log file for your runs (can you attach it) and some more info:

* This is Enterprise ID, yes?  Or, if it's federated, you are using emails to identity people, not usernames?
* In step 2, when you say you "deleted" the user from the dashboard, I assume you mean you went to the Users list and selected the user and hit the minus (-) button, yes?
* In step 3, you mean you changed just the user's First Name, yes?  Not the email?
* After step 4, did you refresh the dashboard to get the user to reappear?  And then you saw his name was still Name1?  And then after the second run you refreshed and it was Name2?

So without the log file I can't be sure, but suspect that the first run after removing the user thought the user was still in the organization (when in fact the user account was in the domain, but the user was not in the organization).  So probably user sync just tried to update the user's first name, and that probably failed with an error message from the server saying it couldn't update the name because the user was not in the org.  However, at the same time, it would have put the user back in the org (possibly because you did some group manipulation as well in the same run).  Then when you ran the second time, because the user was in the org, the update of the first name worked.

Can you attach the log, please, so I can confirm?

**Comment by raymond-ensemble** [comment]
- I'm using a Enterprise ID.
- Yes
- Yes, just the first name
- Yes. After step 2, I confirmed that the user is gone from the dashboard. Then, after step 4, I confirmed that the user is added back to the dashboard, but with Name. After I ran the update user info, it changed to Name2.
I just changed the name and didn't touch the groups. Here's the log:
[2017-03-24.txt](https://github.com/adobe-apiplatform/user-sync.py/files/869553/2017-03-24.txt)

**Comment by adobeDan** [comment]
So there are a lot of firstname updates in the log file :).  Can you tell me which user (email and first name) that you updated?  And what was the name before and what was the name after?  Thanks!

**Comment by raymond-ensemble** [comment]
I was modifying cce7@ensemble.ca. His name initially was "CCE", then I changed it to "CCE_new_name".

**Comment by adobeDan** [comment]
OK, great, thanks, I see it now.  In your log, I see you adding the user with the old name in a run at 18:28:40, then adding him again on the next run with the new name at 18:29:23, then updating his name on the next run at 18:29:41.  So I am guessing you deleted him from the dashboard between 18:28:40 and 18:29:23.

Just to make sure I am completely up-to-speed on the logs:  I also see you adding this user with the old name (`CCE`) earlier at 18:26:12, and then updating him to the *same* name in the next run at 18:26:51.  So what happened there?  Had you edited his name in the dashboard and then set it back to what it was?

I have enough from this that I will do a separate umapi-client test case to figure out if this is a UMAPI server issue or a user-sync issue.  But get back to with verification, please, of what happened in those earlier two runs.

**Comment by adobeDan** [comment]
OK, testing with the umapi_client directly I can see that this is a server-side issue: when you "create" a user by actually re-adding an existing user to the organization, no attempt is made to update the user's attributes as specified in the call, even if the `on_conflict` option is set to `updateIfAlreadyExists`.  I've attached [a log of a run](https://github.com/adobe-apiplatform/user-sync.py/files/871634/user-sync.issue.114.ad-hoc.txt) where we drop a user and re-add him but at the same time we re-add a user that hasn't been dropped.  In the first case, the re-add doesn't update the user; in the second case it does.

I'm going to close this issue as a bug that we won't fix, because we can't fix it in user-sync.  I will file a separate bug inside of Adobe against the server side to see if it can be fixed there.
